torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -366,6 +366,8 @@ def _reset_lazy_init(self) -> None:
         """
         self._is_root: Optional[bool] = None
         self._streams: Dict[str, torch.cuda.Stream] = {}
+        self._fsdp_graph_order: List[nn.Module] = []
+        self._my_fsdp_idx_in_graph: Optional[int] = None
```

Comment:
Can we have a single structure such as Dict[idx, nn.Module]?

Reply:
we need List structure to preserve the order of layers